backend/test/server.cpp

```diff
@@ -16,16 +16,40 @@ TEST_CASE("Test endpoints using handler")
 
     cors.global().headers("*").methods("POST"_method);
 
-    // create a mock password set
-    std::unordered_set<std::string> passwords = password::generatePasswords(3, 12);
-
     // initialize endpoints
     server::root(app);
 
+    // open a database file
+    std::string path = "test.db";
+    std::ofstream file(path);
+    REQUIRE(file.is_open());
+
+    // create the database, with tables passwords and secret
+    database::Database db = database::Database(path);
+    REQUIRE_NOTHROW(db.execute("CREATE TABLE passwords (password TEXT);"));
+    REQUIRE_NOTHROW(db.execute("CREATE TABLE secret (key TEXT);"));
+
+    // create a mock password set
+    std::unordered_set<std::string> passwords = password::generatePasswords(3, 12);
```

Comment:
**suggestion**: Try to not overlap implementation(s) with testing. If you can hardcode a mock dataset, that is much better than using the same code our actual implementation uses. So, create a set with some of your own passwords. Then it's also deterministic.